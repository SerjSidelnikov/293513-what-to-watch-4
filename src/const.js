export const RatingEnum = {
  BAD: `bad`,
  NORMAL: `normal`,
  GOOD: `good`,
  VERY_GOOD: `very good`,
  AWESOME: `awesome`,
};

export const TabsEnum = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const ACTORS_MAX_LENGTH = 4;
export const MORE_LIKE_FILMS = 4;
export const ALL_GENRES = `All genres`;
export const MAX_SHOW_FILM = 8;
