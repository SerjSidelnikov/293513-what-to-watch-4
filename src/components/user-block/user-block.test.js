import React from 'react';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import UserBlock from './user-block';
import NameSpace from '../../reducers/name-space';
import {AuthorizationStatus} from '../../const';

const mockStore = configureStore([]);

describe(`UserBlock`, () => {
  it(`UserBlock rendered correctly (User is logged in)`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <UserBlock/>
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`UserBlock rendered correctly (User is not logged in)`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
      },
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <UserBlock/>
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
