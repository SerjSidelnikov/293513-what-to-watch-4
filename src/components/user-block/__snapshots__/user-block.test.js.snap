// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock UserBlock rendered correctly (User is logged in) 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__avatar"
    href="/myList"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <img
      alt="User avatar"
      height="63"
      src="/img/avatar.jpg"
      width="63"
    />
  </a>
</div>
`;

exports[`UserBlock UserBlock rendered correctly (User is not logged in) 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
